jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -87,4 +97,53 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
             }
         )
     }
+
+    fun checkRequiredPermissions(project: Project, clusterArn: String, taskArn: String) {
+        val ecsClient = project.awsClient<EcsClient>()
+        val iamClient = project.awsClient<IamClient>()
+        val task = ecsClient.describeTasks(DescribeTasksRequest.builder().tasks(taskArn).cluster(clusterArn).build())
+        val taskRoleArn = if (task.tasks()[0].overrides().taskRoleArn() != null) {
+            task.tasks()[0].overrides().taskRoleArn()
+        } else {
+            val roleArn = project.getResourceNow(EcsResources.describeTaskDefinition(task.tasks()[0].taskDefinitionArn())).taskRoleArn()
```

Comment:
What thread is this on?

Reply:
The thread begins here: https://github.com/aws/aws-toolkit-jetbrains/pull/2653/files#diff-dd5452d78ae3e069245b59d2a785ead9ef21710f88f584703405e6295df57bb4R87, and similarly for `RunCommandDialog` . Reason for doing that is exceptions could be caught from within the same thread instead of communicating between 2.